datafusion-cli/src/main.rs

```diff
@@ -142,11 +163,20 @@ pub async fn main() -> Result<()> {
 }
 
 fn create_runtime_env() -> Result<RuntimeEnv> {
+    let mut table_factories: HashMap<String, Arc<dyn TableProviderFactory>> =
+        HashMap::new();
+    table_factories.insert("csv".to_string(), Arc::new(ListingTableFactory::new(FileType::CSV)));
+    table_factories.insert("parquet".to_string(), Arc::new(ListingTableFactory::new(FileType::PARQUET)));
+    table_factories.insert("avro".to_string(), Arc::new(ListingTableFactory::new(FileType::AVRO)));
+    table_factories.insert("json".to_string(), Arc::new(ListingTableFactory::new(FileType::JSON)));
```

Comment:
@alamb this is getting closer to something I think you asked for earlier: treating `ListingTables` and custom tables the same way and just registering factories for table types.